lib/job/run.js

```diff
@@ -60,11 +67,19 @@ module.exports = function() {
       }
       if (definition.fn.length === 2) {
         debug('[%s:%s] process function being called', self.attrs.name, self.attrs._id);
-        definition.fn(self, jobCallback);
+        try {
+          await definition.fn(self, jobCallback);
```

Comment:
The arity check is to see whether the job processor function accepts a callback. I don't see any need to support accepting a callback *and* returning a promise